hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -173,6 +176,7 @@
         this.operationProviders = createOperationProviders();
         this.partitioningStrategyFactory = new PartitioningStrategyFactory(nodeEngine.getConfigClassLoader());
         this.nodeWideUsedCapacityCounter = new NodeWideUsedCapacityCounter(nodeEngine.getProperties());
+        this.loadedKeyLimiter = new Semaphore(nodeEngine.getProperties().getInteger(LOADED_KEY_LIMITER));
```

Comment:
Here we need to validate a positive value of `nodeEngine.getProperties().getInteger(LOADED_KEY_LIMITER)`, otherwise the map loader will be stuck

Reply:
done